csrc/python_frontend/python_bindings.cpp

```diff
@@ -576,40 +570,47 @@ void initNvFuserPythonBindings(PyObject* module) {
   // This is the input version of define_vector
   fusion_def.def(
       "define_vector",
-      [](FusionDefinition& self,
-         size_t size,
-         PrimDataType dtype = DataType::Int) -> Vector {
-        return define_vector_fn(self, std::nullopt, size, dtype);
-      },
-      py::arg("size"),
-      py::arg("dtype") = DataType::Int,
-      py::return_value_policy::reference);
-  // This is the canonical version of define_vector that accepts either a
-  // nullptr or a vector of values to indicate either an input or a constant
-  // for use when printing out the associated Fusion Record.
-  fusion_def.def(
-      "define_vector",
-      [](FusionDefinition& self,
-         std::optional<std::vector<int64_t>> value,
-         size_t size,
-         PrimDataType dtype) -> Vector {
-        return define_vector_fn(self, value, size, dtype);
+      [](FusionDefinition& self, size_t size) -> Vector {
+        std::vector<Scalar> args;
+        args.reserve(size);
+        for (size_t i = 0; i < size; ++i) {
+          Scalar out = self.defineScalar();
+          self.defineRecord(new ScalarRecord<int64_t>(
+              {self.recordingState(out())},
+              serde::RecordType_ScalarInput,
+              std::nullopt,
+              DataType::Int));
+          args.emplace_back(out);
+        }
+        return define_vector_fn(self, args);
       },
-      py::arg("value").none(true),
       py::arg("size"),
-      py::arg("dtype") = DataType::Int,
       py::return_value_policy::reference);
   // This is the constant version of define_vector when given a vector
   // of constant values.
   fusion_def.def(
       "define_vector",
-      [](FusionDefinition& self,
-         std::vector<int64_t> value,
-         PrimDataType dtype) -> Vector {
-        return define_vector_fn(self, value, value.size(), dtype);
+      [](FusionDefinition& self, std::vector<int64_t>& value) -> Vector {
```

Comment:
Will it eventually be possible to have a combined vector with both constants and inputs? For example if the user wants to do `pad(tv, [0, s1])`

Reply:
Or even more commonly for reshapes where one entry is -1 and others are based on an input extent.

Reply:
By that I mean not having to construct a Scalar manually for 0 or -1, but accepting a list of python variables in a single signature and unwrapping them there to create constants if needed.

Reply:
I am not sure.   If it is useful enough, we could make it.  I just don't know what the framework will do.